search.go

```diff
@@ -282,9 +282,16 @@ type SearchRequest struct {
 	SearchAfter      []string          `json:"search_after"`
 	SearchBefore     []string          `json:"search_before"`
 
+	Similarity *SimilarityRequest `json:"similarity"`
```

Comment:
Is Similarity an appropriate name?

Reply:
Let's use kNN everywhere - KNNSearcher, kNNMatch etc.